cli/azd/pkg/commands/pipeline/pipeline_manager.go

```diff
@@ -420,3 +446,42 @@ func (manager *PipelineManager) Configure(ctx context.Context) (
 		PipelineLink:   strings.TrimSuffix(ciPipeline.remote, ".git"),
 	}, nil
 }
+
+func checkRoleAssignments(
+	ctx context.Context,
+	azcli azcli.AzCli,
+	subscriptionId string,
+	principalId string,
+	console input.Console,
+) error {
+	// Find the specified role in the subscription scope
+	scope := azure.SubscriptionRID(subscriptionId)
+	roles, err := azcli.GetUserRoleDefinitionName(ctx, subscriptionId, scope, principalId)
```

Comment:
With this current implementation there are going to be other edge case conditions that we are not accounting for.  Users could have the correct permissions but may be inherited via custom role definition or roles that have been assigned to a group that the user is a member of.

This could lead to the user getting the confirmation prompt (causing confusion) but then ignoring and continuing would actually work.

We should consider getting the effective permissions of the user and checking the individual permission.  I'll look to see if there is an API for this and get back to you.

Reply:
> We should consider getting the effective permissions of the user and checking the individual permission. I'll look to see if there is an API for this and get back to you.

Interested in seeing how this works out.  I agree that this is the correct check, but it was non obvious to me how to actually make it.  Interested if you can dig something up here, @wbreza.

The other long term option would be for us to take the error returned by the role assignment operation and examine it and then decide if the error was due to invalid permissions and guide the user on how to get a role assignment setup for them to work.

But I do think we capture about 90% of the value with this change as is, and I wouldn't block landing it here if it takes us a while to figure out how to ask ARM "does this principal have this permission on this resource".